AndroidHomeApplication/app/src/main/java/com/example/androidhomeapplication/observer/DataResult.kt

```diff
@@ -0,0 +1,22 @@
+package com.example.androidhomeapplication.observer
+
+
+class DataResult<T> {
```

Comment:
use sealed class

sealed class Result<out T : Any?> {
    class Loading : Result<**Nothing**>()
    class Success<out **T : Any**>(val value: T) : Result<**T**>()
    class Error(val error: Throwable) : Result<**Nothing**>()
}

and handle it using when